src/units/unit.cpp

```diff
@@ -2428,6 +2428,33 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 			level_ += lexical_cast_default<int>(increase);
 		}
 	}
+
+	// In case the effect carries EventWML, apply it now
+	if(resources::game_events && resources::lua_kernel) {
+		config events;
+		for(const config& effect_event : effect.child_range("event")) {
+			events.add_child("event", effect_event);
+		}
+		for(const config& abilities : effect.child_range("abilities")) {
```

Comment:
Can we limit this to `if(effect["apply_to"] == "new_ability")`? For example, it shouldn't be adding events if the effect was to _remove_ an ability; and in the case of a custom effect, it should be left up to that effect whether it wants to insert an event. (Maybe the Lua unit/event API will need extending to allow inserting an event directly into a unit.)